### PR TITLE
Fixes to GWAS General Tutorial

### LINKED ISSUE #122: Add tech/credit prerequisites and mention AWS time
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-Cloud/

- [x] this tutorial is written for a Mac, but does not have any technology requirements listed. First mention that I need a mac is the accessing_aws page: "OK, so you've created a running computer on the cloud. How do you get to it? We will use the Mac terminal window to access the cloud computer."
- [x] It also doesn't tell me I need a credit card (which I'm pretty sure I do to set up an AWS account regardless of whether I only use free compute)
- [x] does not include the "waiting for AWS to approve me" that can take anywhere from minutes to days. That should all be in here.

### LINKED ISSUE #123: Choose and implement a single AWS configuration
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-Cloud/aws_instance_setup/

a t2.micro instance for Ubuntu Server 20.04 LTS (HVM), SSD Volume Type - ami-0d57c0143330e1fa7 (64-bit x86) / ami-05f126563b93d66c6 (64-bit Arm) cannot be launched without configuring a VPC if you are in US east (N. Virgina) us-east-1. 
image
Need to tell uses a specific region to use. us-east-1, us-west-1 and us-west-2 all seem to have the machine config you picked. However, it's very likely the the pre-installed software differs across regions,

I would pick *one* region, one AMI, one size, etc and tell your user to just use that one rather than give them options, then modify the lesson to fit that one instance.

### LINKED ISSUE #127: Fix wording
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-Cloud/Accessing_aws/

- [x] "Start this tutorial by opening up a terminal window by searching (type cmd+space_bar) for "terminal" on your Mac." implies that I am now doing a new tutorial. Change to something like "To access AWS you need to open a terminal..."

- [x] This sentence is super confusing: "Copy this name, connect to the cloud computer with ssh under the username ‘ubuntu’, as follows:" Especially since the very next thing you say to do is chmod a file that does not seem to have anything to do with that sentence. The flow of a tutorial should be "Our goal is X. To do X, we need Y. These are the steps of Y". What you have is "Do Y. Do Z. This is Y."

### LINKED ISSUE #128: clarify chmod
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-Cloud/Accessing_aws/

what is chmod? why am I doing it? what are these letters? There should be at least a short explanation. I generally also include a link to the [wikipedia page](https://en.wikipedia.org/wiki/Chmod)

In Amazon, once your instance is running if you click the 'connect' button, it gives you personalized to your instance instructions for logging into that instance:
image
 Yours is different. Is it *functionally* different? Users who don't know chmod won't know what is happening here.

### LINKED ISSUE #168: Typos in GWAS lesson
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-cloud/background/

> Researchers identify two groups: one groups of people with a disease condition (patients) and one group of healthy individuals (controls).

should be 'one group of people'

> Next, genomes of diseases individuals are compared to the genomes

should be 'of diseased individuals'

> For example, large number of associated SNPS in the gene encoding eye color may not serve as a good candidate gene for liver disease. These SNPs may just be "tagging along" with the actual causal variants.

So there's a typo in there, but I also think it's just conflating a bunch of different issues. There's no reason an eye color gene couldn't be implicated in liver disease, the concern is more about the specificity of a correlation analysis given that genomes tend to have large haplotype blocks. I would just change this to:

"For example, you may find multiple SNPs in a region that are equally well correlated with liver disease, but that only one of those SNPs cause the phenotype. It's also common to find that none of the correlated SNPs are causative, rather they all are just tightly linked to an undetected mutation and are "tagging along" with the actual causal variants.

### LINKED ISSUE #169: GWAS install streamlining
In general, you don't want to ruin the flow of your tutorial with breaking to watch things install all the time. Find a point near the beginning where you Install everything on AWS at once. In an in-person lesson, you can use that install time to give them an overview of "we're installing this stuff, they do these steps, and that makes a pipeline" You do the talk while the installs run forever. For the online version, you should make a little flowchart image that shows the steps for them to look at while stuff installs. Basically, you have a nice 'what is GWAS' conceptual overview, but you don't have anything right now to orient your user to the workflow. They're just installing things and typing things when you tell them to, without any way of telling where they are in the pipeline. 

## specific install issues:

in https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-cloud/vcftools_install/

This code block is super confusing and repetitive. 

```
sudo apt-get install autoconf autogen
sudo apt-get update
sudo apt-get install g++
sudo apt install gcc make autoconf automake pkg-config zlib1g-dev curl
sudo apt-get update
sudo apt-get install pkg-config
```
You can't install autogen before you do the apt-get update, it just fails. You're installing autoconf and pkg-config twice and there's no need to keep running `sudo apt-get update` 

I would move all of the installs to where you are installing `wget`, run `sudo apt-get update` *first* then install the other packages you need, once each. And it would be easier if you did unzip as your explained example and then run all the rest as a single line so something like:

new lesson flow
-------

we need to make sure our app installer is up to date because we're using the LTS whatever version which is frozen at the version it says, and so always needs to be updated when we start:

`sudo apt-get update`

Now we can install stuff. one thing we need is unzip. blah blah

`sudo apt-get install unzip`

blah blah we can check it's installed by looking at the help

`unzip -h`

We actually need lots of helper utilities to run todays pipeline so lets install them all now:

`sudo apt-get install autoconf autogen g++ gcc make automake pkg-config zlib1g-dev curl gdebi-core`

while we wait, let's go over the workflow we're going to use. We're gonna get data, then use PLINK to...

### LINKED ISSUE #170: Manhattan plot code error
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-cloud/manhattan/

the running the plot function code just doesn't work. I had to copy it all out and make it a single line in a text editor to make it function. It looks like this rscript can't accept / line breaks

- In the next step, it says  "The output should look like this:" but, as long as you give it a valid data file, you get exactly that output regardless of whether the program works or not (so, the files might all be empty, but there). Change your check to `ls -lthr`, which will show the last file made, and it's size. So, the user should check to see that they have a 34k file called coatColor_man.bmp

- for visualization, unless you have a reason for them to download all those files, switch to just downloading the bmp and make your note about how to change it to download everything. Otherwise you are likely to crash your classroom internet connection.

### LINKED ISSUE #171: Move termination
https://cfde-training-and-engagement.readthedocs-hosted.com/en/latest/Bioinformatics-Tutorials/GWAS-in-the-cloud/terminate_aws/

The `important` should be first, not last. People tend to do the instructions *THEN* read the notes while they wait for a thing to happen. They *will* just delete it without looking to see that's potentially bad.

### LINKED ISSUE #172: Analyze page needs more context
# Analyze

> In order to specify the minor allele as the reference allele for PLINK

 - [ ] Why do I want to specify minor alleles?  

- [x] So. There are certainly times where you don't want to explain all the code. AWK is absolutely one of those times. We aren't here to teach people AWK, and we don't have the 20 years it would take to do it. So that's fine. But, it's still best not to just drop complex code on people with no explanation at all. A couple sentences about what this code philosophically does, a throwaway about how learning AWK is a thing for another day, and a link to a nice AWK page should be here. Personally, I always link to grymoire for AWK and SED, because he actually does a pretty good job of making them approachable, and his writing style is really close to mine, so it fits really well in my tutorials [awk](https://www.grymoire.com/Unix/Awk.html), [sed](https://www.grymoire.com/Unix/Sed.html) 
- The plink note is good and helpful :)
- [x] What does `Test value is [yellow] and missing value is [-9]` and `53 of 53 individuals assigned to 2 cluster(s)` mean?
- [x] I would not just assume that people know how to quit `less` here
- [x] I think it's okay to have a very basic tutorial like this that doesn't show all the options, but I think that a second tutorial that *does* show how to exclude SNPs and individuals would be very, very helpful. Those are extremely common things to need to do, and no one with real data would be able to do a GWAS without those steps, so adding them gives the user the ability to *practically* apply these new skills. 
- As an aside, the fact that you have said this is the most simple tutorial, and are not showing options underscores that the audience you're aiming at is *newbies*, so everything needs the amount of context and info that you have for the GWAS interpretation stuff here.

### LINKED ISSUE #173: Download the data errors
- [x] `wget` is not installed on macs by default. Since this is a mac-centric lesson, you can't have people use wget unless you teach them to install it locally. Generally, you don't want to make them install stuff locally, it's much more error prone than AWS. And, with AWS if they ruin it they can kill the instance and start over...if they break their local machine...its not good. Ask me about the time one of the ANGUS instructors accidentally made someone delete their local operating system. You should be using curl. `curl -LO` will download the output using the name on the server, and follow any symbolic links it needs to
- [x] you need to explain what SCP is and how they know this is how the command looks. Otherwise they will never be able to do this with their own data

### PR DESCRIPTION
**Pull request by abhijna**
## PR Checklist

- [x] Release label added
- [x] PR category label added
- [x] PR mergeable
- [x] Clean build logs
- [x] Linked related issues
- [ ] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description
Issues: #170 #169 #128 #127 #123 #173 #168 #171 #172
This PR address all the issues Amanda had with the GWAS General tutorial including clarifications, explanations, and fixes to the code and text, streamlining of installations, fixes to typos, and reorganization of certain sections. One issue that Amanda had with Manhattan plots R code is not an issue for me #170. Please copy+paste code block as is and let me know if you can run the code. The line breaks don't seem to matter. 
AWS changed their ec2 user interface recently. I've added an admonition to the top of the tutorial. This may not be the best location for it, so if you want it to be moved, please let me know. 

## Review format
Please leave all suggestions and proposed fixes (including typos) as comments in the PR and I will fix them.

## Timeline
This has an Oct 2020 tag. Please provide comments back with at least 1-2 days' time for me to incorporate your changes.

## Reviewer Checklist

- [ ] spelling and grammar
- [ ] successful run of all installation and code chunks
- [ ] sufficient explanation and details for the tutorial content
- [ ] code syntax and naming convention
- [ ] adherence of tutorial format to style guide
- [ ] functional links (inter and intra)
- [ ] accessible hyperlink text

### CODE REVIEW

**Review by marisalim — CHANGES_REQUESTED** [review]
Looking good @abhijna! Here are my suggestions. Let me know if you need clarification!

from: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/download_accessAWS/#install-other-helper-utilities
- [x] this line of code is VERY important, all sorts of errors in installations/plotting happen if it's not run! could you add a note on that? for example, the vcftools installation will fail because it needs autoconf, autogen, and make; Rscript plotting will fail because it needs ghostscript. this might be a good place to explain more about some of what these helper utilties are needed for.
```
sudo apt-get install autoconf autogen g++ gcc make automake pkg-config zlib1g-dev curl gdebi-core -y ghostscript-x
```

this PR addresses several issues, here are my comments by issue:

## Add tech/credit prerequisites and mention AWS time #122
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/
- [x] typo:
> AWS account set up needs approval by AWS, and approval times can **range** from minutes to days.
- [x] change the estimated time format to e.g., 10 mins instead of 00:10

## Choose and implement a single AWS configuration #123
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/aws_instance_setup/
- [x]  i would rephrase to:
> **For this tutorial, it** is important to select the **"Ohio"** amazon machine image. The geographical region of your remote machine is displayed on the top right of this page (shown in image below)."

## Fix wording #127
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/download_accessAWS/
- [x] edit title: this page is titled "Download and move data to AWS" but "Download Data & Install Helper Utilities" in the left TOC
-[x] update format:
> To download data onto your local computer you need to open up a terminal window. You can do this by searching (type `cmd+space bar`) for "terminal" on your Mac.
-[x] could you change the javascript tab titles to:
```
=== "Local machine"
    
    for code on local machine

=== "Expected Output"

    (for the ones that you included output for)

=== "AWS instance"

    for code on AWS instance

=== "Expected Output"
```

- [x] format suggestion:
```
You will see this message when running the `ssh`
command for the first time:

> The authenticity of host 'ecc2-???-???-???-???.compute-1.amazonaws.com (3.129.57.169)' can't be established. ECDSA key fingerprint is XXX. Are you sure you want to continue connecting (yes/no/[fingerprint])?

Type "yes" and press `ENTER`.
```

## clarify chmod #128
- lgtm

## Typos in GWAS lesson #168
- lgtm

## GWAS install streamlining #169
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/vcftools_install/
-[x] for the software installation steps, can you add a command to show whether the installation was successful, like checking for the software version:
```
=== "AWS instance"

    plink -h
    
=== "Expected Output"

    The top portion of the help page:
    @----------------------------------------------------------@
    |        PLINK!       |     v1.07      |   10/Aug/2009     |
    |----------------------------------------------------------|
    |  (C) 2009 Shaun Purcell, GNU General Public License, v2  |
    |----------------------------------------------------------|
    |  For documentation, citation & bug-report instructions:  |
    |        http://pngu.mgh.harvard.edu/purcell/plink/        |
    @----------------------------------------------------------@


    Please visit the PLINK website for a complete list of options

    A few common options are listed here:

    plink --file {fileroot}         Specify .ped and .map files 
      --bfile {fileroot}        Specify .bed, .fam and .map 

      --out {fileroot}          Specify output root filename  
```
- [x] typo:
> The first line of code changes **your** directory to the home directory
- [x] the vcftools installation works as you wrote it, but just a note that if for some reason, the user, hadn't ahem done the download data step, they would need to run through the autoconf, etc. code. oops :)
```
cd
git clone https://github.com/vcftools/vcftools.git
cd vcftools

# i needed this section of code for autogen, etc. 
sudo apt-get update
sudo apt-get install unzip
sudo apt-get install autoconf autogen g++ gcc make automake pkg-config zlib1g-dev curl gdebi-core

./autogen.sh
./configure
make
sudo make install
```

## Manhattan plot code error #170
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/manhattan/
- [x] plotting code works!
- [x] could you add a short explanation for this notation: MC1R (c.916C>T)
- [x] edit suggestion:
> You can visualize the Manhattan plot by downloading the **"coatColor_man.bmp"** file to our local computers
> Watch this video for **more** explanation of GWAS Manhattan plots.
- [x] add a reminder to replace the ec2 part to your particular instance, it won't work if they directly copy/paste!
> Now run the following code on your Mac terminal window (not Ubuntu!) to do the actual copying:


## Move termination #171
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/terminate_aws/
- [x] maybe add that it's important to terminate the instance, otherwise AWS will keep on charging you since the instance is still running
- [ ] some kind of Key Points or conclusion would be nice, since this is a longer tutorial - could be a recap of the learning objectives

## Analyze page needs more context #172
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/analyze/
- [x] are these supposed to be listed out 1 per line? -- nope just like that
```
Per-individual QC of GWA data consists of identifying individuals with: 1) missing genotype or heterozygosity rate. 2) replicated samples or closely related individuals. 3) identification of individuals of divergent ancestry.

Per-marker QC of GWA data consists of identifying SNPs with: 1) lots of missing genotypes. 2) significant deviation from Hardy-Weinberg equilibrium (HWE). 3) largely different missing genotype rates between cases and controls. 4) very low minor allele frequencies
```
- [x] format suggestion: move the input and output javascript next to each other for the plink command, so that users toggle between the tabs to see code. this command:
```
plink --file coatColor --make-pheno coatColor.pheno "yellow" --missing --out miss_stat --noweb --dog --reference-allele minor_alleles --allow-no-sex --adjust
```
- [x] typo:
> This means 53 individuals have been assigned to two clusters: test (yellow coat color) and control (dark coat color). **There** 
- [x] format edit: can you put the flags in backticks (there are a few other places on this page) and files in ""
> `--mind` or `--geno` or `--maf`.
> The per individual and per SNP rates are then output to the files "miss_stat.imiss" and "miss_stat.lmiss", respectively. If you had not specified an `--out` option,
- [x] format edit: change "here" to description or put the hyperlink on the flag text itself
> --make-bed: creates a new PLINK binary file set, after applying sample/variant filters and other operations. Click here for more details

## Download the data errors #173 
comments for: https://cfde-training-and-engagement--218.com.readthedocs.build/en/218/Bioinformatics-Tutorials/GWAS-in-the-cloud/download_accessAWS/
- [ ] i can directly download the data with the curl commands to my AWS instance terminal. that streamlines the steps on this page a lot, and you don't need to do the local download and scp transfer steps. i opened the instance with the ssh command (your step 3), and made GWAS directory, and ran curl commands (your step 1, but in AWS).

**Comment by abhijna** [comment]
thanks for the detailed comments @marisalim. Where would you like the conclusions to live? As a last page? I made almost all the changes. I still can't download cyverse data onto AWS. This is the error I get:
```
57-9C3A-BAEAA0CCC694/pruned_coatColor_maf_geno.vcf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:02:11 --:--:--     0
curl: (28) Failed to connect to de.cyverse.org port 443: Connection timed out
```

**Comment by marisalim** [comment]
@abhijna the conclusion and/or simple Key Points section can go on the last pg of your tutorial, it doesn't have to be it's own separate page.

Interesting, I'll check on the curl error tmw!
